include/sampling/random_point_generators_multithread.hpp

```diff
@@ -0,0 +1,262 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+
+// Contributed and/or modified by Konstantinos Pallikaris, as part of Google Summer of Code 2021 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef SAMPLERS_RANDOM_POINT_GENERATORS_MULTITHREAD_HPP
+#define SAMPLERS_RANDOM_POINT_GENERATORS_MULTITHREAD_HPP
+
+#include <iostream>
+#include <omp.h>
+#include <unistd.h>
+
+
+template <typename GenericWalk>
+struct policy_storing
+{
+    template <typename WalkPolicy, typename PointList, typename ThreadParameters>
+    static void store(WalkPolicy &policy, PointList &randPoints, ThreadParameters &thread_random_walk_parameters) 
+    {
+        policy.apply(randPoints, thread_random_walk_parameters.p);
+    }
+};
+
+
+template <>
+struct policy_storing<BRDHRWalk_multithread>
+{
+    template <typename WalkPolicy, typename PointList, typename ThreadParameters>
+    static void store(WalkPolicy &policy, PointList &randPoints, ThreadParameters &thread_random_walk_parameters) 
+    {
+        policy.apply(randPoints, thread_random_walk_parameters.p1);
+        policy.apply(randPoints, thread_random_walk_parameters.p2);
+    }
+};
+
+template <>
+struct policy_storing<BCDHRWalk_multithread>
```

Comment:
you do not have to repeat it here, just inherit the method from `policy_storing<BRDHRWalk_multithread>` i.e. 
```
template <>
struct policy_storing<BCDHRWalk_multithread> : policy_storing<BRDHRWalk_multithread> 
{}
```

Reply:
Thank you @vissarion. I've implemented this change.